src/transformations/molodensky.cpp

```diff
@@ -146,10 +146,20 @@ static PJ_LPZ calc_standard_params(PJ_LPZ lpz, PJ *P) {
     dphi  = (-dx*sphi*clam) - (dy*sphi*slam) + (dz*cphi)
             + ((nu * P->es * sphi * cphi * da) / a)
             + (sphi*cphi * ( rho/(1-f) + nu*(1-f))*df);
-    dphi /= (rho + lpz.z);
+    const double dphi_denom = rho + lpz.z;
+    if( dphi_denom == 0.0 ) {
+        lpz.phi = HUGE_VAL;
+        return lpz;
+    }
+    dphi /= dphi_denom;
 
     /* delta lambda */
-    dlam = (-dx*slam + dy*clam) / ((nu+lpz.z)*cphi);
+    const double dlam_denom = (nu+lpz.z)*cphi;
+    if( dlam_denom == 0.0 ) {
+        lpz.phi = HUGE_VAL;
```

Comment:
It probably doesn't matter in the end, but shouldn't it rather be `lpz.lam = HUGE_VAL` here?

Reply:
Done